tests/unit/lib/reports/KinnexReport.spec.js

```diff
@@ -0,0 +1,377 @@
+    setActivePinia(pinia)
+    // Mock the root store and its API
+    rootStore = useRootStore()
+    get = vi.fn()
+    rootStore.api.traction.pacbio.requests.get = get
+
+    ssGet = vi.fn()
+    rootStore.api.sequencescape.studies.get = ssGet
+  })
+
+  describe('fetchTractionSamples', () => {
+    it('returns empty array and shows alert if both responses fail', async () => {
+      get.mockRejectedValue('Internal Server Error')
+
+      const { data, errors } = await fetchTractionSamples('input', [])
+      expect(data).toEqual([])
+      expect(errors).toEqual({
+        message: 'Error fetching samples from Traction: Internal Server Error',
+        type: 'danger',
+      })
+    })
+
+    it('returns empty array and shows alert if no data found', async () => {
+      get.mockResolvedValue({
+        status: 200,
+        statusText: 'OK',
+        ok: true,
+        json: () => Promise.resolve({ data: [], included: [] }),
+      })
+
+      const { data, errors } = await fetchTractionSamples('input', [])
+      expect(data).toEqual([])
+      expect(errors).toEqual({
+        message: 'No samples found in Traction with the provided input',
+        type: 'warning',
+      })
+    })
+
+    it('removes duplicates and returns formatted samples', async () => {
+      // Duplicates because the same sample is found in both requests
+      get.mockResolvedValue({
+        status: 200,
+        statusText: 'OK',
+        ok: true,
+        json: () => Promise.resolve({ data: [request], included: [sample] }),
+      })
+
+      const { data, errors } = await fetchTractionSamples('input', [])
+      expect(errors).toEqual({})
+      expect(data).toHaveLength(1)
+      expect(data[0]).toEqual({
+        request_id: request.id,
+        barcode: request.attributes.barcode,
+        cost_code: request.attributes.cost_code,
+        library_type: request.attributes.library_type,
+        external_study_id: request.attributes.external_study_id,
+        date_of_sample_collection: sample.attributes.date_of_sample_collection,
+        supplier_name: sample.attributes.supplier_name,
+        donor_id: sample.attributes.donor_id,
+        species: sample.attributes.species,
+        sanger_sample_id: sample.attributes.sanger_sample_id,
+      })
+    })
+
+    it('successful if only one of the requests returns data', async () => {
+      get.mockResolvedValueOnce({
+        status: 200,
+        statusText: 'OK',
+        ok: true,
+        json: () => Promise.resolve({ data: [request], included: [sample] }),
+      })
+      // Second call is for source_identifier fetch
+      get.mockResolvedValueOnce({
+        status: 200,
+        statusText: 'OK',
+        ok: true,
+        json: () => Promise.resolve({ data: [], included: [] }),
+      })
+
+      const { data, errors } = await fetchTractionSamples('input', [])
+      expect(errors).toEqual({})
+      expect(data).toHaveLength(1)
+      expect(data[0]).toEqual({
+        request_id: request.id,
```

Comment:
same as above. You can do something like `expect(data[0]).toEqual(formattedSample({sample, request})`